datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/JavaTimeModule.java

```diff
@@ -142,6 +144,14 @@ public void setupModule(SetupContext context) {
         desers.addDeserializer(ZoneOffset.class, JSR310StringParsableDeserializer.ZONE_OFFSET);
 
         context.addDeserializers(desers);
+
+
+        boolean oneBasedMonth = _features.isEnabled(JavaTimeFeature.ONE_BASED_MONTHS);
+        Supplier<Boolean> serializeEnumsByIndex = () -> context.isEnabled(SerializationFeature.WRITE_ENUMS_USING_INDEX);
```

Comment:
Let's not do this: setting should be accessed from either `SerializerProvider` or `SerializerConfig`: `SetupContext` is not guaranteed to be valid beyond time Modules are registered -- and adding a closure will keep a strong reference to it.